test/widgets/message_list_test.dart

```diff
@@ -922,6 +922,72 @@ void main() {
         await tester.pump();
         tester.widget(find.text('new stream name'));
       });
+
+      testWidgets('navigates to TopicNarrow on tapping topic in CombinedFeedNarrow', (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+        final channel = eg.stream();
+        final message = eg.streamMessage(stream: channel, topic: 'topic name');
+
+        await setupMessageListPage(
+          tester,
+          narrow: const CombinedFeedNarrow(),
+          messages: [message],
+          subscriptions: [eg.subscription(channel)],
+          navObservers: [navObserver],
+        );
+
+        assert(pushedRoutes.length == 1);
+        pushedRoutes.clear();
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          messages: [message],
+          foundOldest: true,
+        ).toJson());
+
+        final topicFinder = find.descendant(
+          of: find.byType(StreamMessageRecipientHeader),
+          matching: find.text('topic name'),
+        );
+        await tester.tap(topicFinder);
+        await tester.pumpAndSettle();
+
+        check(pushedRoutes).single
```

Comment:
nit: put the check immediately after taking the actions that set up the situation it's checking:
```suggestion
        await tester.pumpAndSettle();
        check(pushedRoutes).single
```

That's helpful for making the test more compact as discussed above, but also helpful for grouping the test's logic to show its structure: this way each stanza separated by blank lines can be some setup steps followed by the corresponding check.